x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/search/BasicQueryClient.java

```diff
@@ -64,48 +68,52 @@ protected void search(SearchRequest search, ActionListener<SearchResponse> liste
     }
 
     @Override
-    public void get(Iterable<List<HitReference>> refs, ActionListener<List<List<GetResponse>>> listener) {
-        MultiGetRequestBuilder requestBuilder = client.prepareMultiGet();
-        // no need for real-time
-        requestBuilder.setRealtime(false)
-                      .setRefresh(false);
+    public void fetchHits(Iterable<List<HitReference>> refs, ActionListener<List<List<SearchHit>>> listener) {
+        IdsQueryBuilder idsQuery = idsQuery();
 
-        int sz = 0;
+        int innerListSize = 0;
+        Set<String> indices = new HashSet<>();
+
+        // associate each reference with its own
+        final Map<HitReference, List<Integer>> referenceToPosition = new HashMap<>();
+        int counter = 0;
 
         for (List<HitReference> list : refs) {
-            sz = list.size();
+            innerListSize = list.size();
```

Comment:
is the `innerListSize` updated correctly? From what I understand it's the last one from the for loop.

Reply:
Correct because each sequence has the same number of entries. The initialization happens in the loop since `refs` is an iterator so it's not possible to just peek at one entry, the whole iterator has to be consumed hence why it's part of the loop.